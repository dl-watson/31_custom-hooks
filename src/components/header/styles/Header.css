.Header {
display: flex;
flex-direction: column;
justify-content: center;
align-items: center;
padding: 40px;

    button {
        padding: 20px;
    }
}