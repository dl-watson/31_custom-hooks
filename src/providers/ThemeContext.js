import React, { useContext, useState } from "react";
import PropTypes from "prop-types";

const ThemeContext = React.createContext();

export const useTheme = () => {
  return useContext(ThemeContext);
};

export const ThemeProvider = ({ children }) => {
  const [darkTheme, setDarkTheme] = useState(false);

  const themeStyles = {
    backgroundColor: darkTheme ? "#444b4d" : "#d4d0cf",
    color: darkTheme ? "#d4d0cf" : "#444b4d",
  };

  const toggleTheme = () => {
    setDarkTheme((prevTheme) => !prevTheme);
  };

  return (
    <ThemeContext.Provider value={{ darkTheme, themeStyles, toggleTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

ThemeProvider.propTypes = {
  children: PropTypes.any,
};
