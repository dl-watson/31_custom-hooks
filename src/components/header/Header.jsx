import React from "react";
import { useTheme } from "../../providers/ThemeContext";
import styles from "./styles/Header.css";

const Header = () => {
  const { themeStyles, toggleTheme } = useTheme();

  return (
    <header className={styles.Header} style={themeStyles}>
      <button onClick={toggleTheme}>Toggle Theme</button>
    </header>
  );
};

export default Header;
