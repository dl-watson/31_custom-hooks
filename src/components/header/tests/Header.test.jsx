global.fetch = require("node-fetch");
import React from "react";
import { render, cleanup } from "@testing-library/react";
import Header from "../Header";
import { ThemeProvider } from "../../../providers/ThemeContext";

describe("Header component", () => {
  afterEach(() => cleanup());
  it("renders Header", () => {
    const { asFragment } = render(
      <ThemeProvider>
        <Header />
      </ThemeProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
