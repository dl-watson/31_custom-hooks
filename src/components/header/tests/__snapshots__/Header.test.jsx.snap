// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header 1`] = `
<DocumentFragment>
  <header
    class="Header"
    style="background-color: rgb(212, 208, 207); color: rgb(68, 75, 77);"
  >
    <button>
      Toggle Theme
    </button>
  </header>
</DocumentFragment>
`;
